File: src/gsqa.gheart.tests.java/src/test/java/gsqa/gheart/tests/java/test/SuiteFalabella.java
package gsqa.gheart.tests.java.test;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.TestMethodOrder;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import gsqa.gheart.sdk.java.helpers.GreenHeartHelper;

@ExtendWith(SuiteFalabellaExtension.class)
@TestMethodOrder(MethodOrderer.Alphanumeric.class)
public class SuiteFalabella extends SuiteFalabellaExtension {

	@DisplayName("VALIDAR_DISPONIBILIDAD_FALABELLA")
	@Test
	public void A_validarDisponibilidadFalabella() {
		driver.navigate().to("https://www.falabella.com/");
		driver.manage().window().maximize();
		GreenHeartHelper.takeEvidenceWithScreenshot("Navegación", "Se navega al sitio web corretamente", driver);
	}

	@DisplayName("BUSQUEDA_FALABELLA")
	@Test
	public void B_busquedaFalabella() throws Exception {
		String text = "Laptop table";
		driver.findElement(By.id("testId-SearchBar-Input")).click();
		driver.findElement(By.id("testId-SearchBar-Input")).sendKeys(text);
		driver.findElement(By.cssSelector("#testId-search-wrapper > div > button")).click();
		Thread.sleep(1000);
		GreenHeartHelper.takeEvidenceWithScreenshot("Búsqueda", "Se realiza la búsqueda de " + text + " correctamente",
				driver);
	}
}

File: src/gsqa.gheart.tests.dotnet/gsqa.gheart.tests.dotnet.fx/SuiteFalabella.cs
using Gsqa.GHeart.Sdk.FX.Helpers;
using Gsqa.GHeart.Sdk.FX.Models;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace gsqa.gheart.tests.dotnet.fx
{
  public class SuiteFalabella
  {
    private static IWebDriver driver;
    private static DateTime startTime;

    [OneTimeSetUp]
    public void TestInitialize()
    {
      driver = new ChromeDriver();
      GreenHeartHelper.Init(false, "123456", "GSQA", "123456", "GreenHeart", "001", "Suite_Falabella", "1.0");
    }

    [SetUp]
    public void Start()
    {
      startTime = DateTime.Now;
    }

    [Test]
    public void A_validarDisponibilidadFalabella()
    {
      driver.Navigate().GoToUrl("https://www.falabella.com/");
      driver.Manage().Window.Maximize();
      GreenHeartHelper.TakeEvidenceWithScreenshot("Navegación", "Se navega al sitio web corretamente", driver);
    }

    [Test]
    public void B_busquedaFalabella()
    {
      String text = "Laptop table";
      driver.FindElement(By.Id("testId-SearchBar-Input")).Click();
      driver.FindElement(By.Id("testId-SearchBar-Input")).SendKeys(text);
      driver.FindElement(By.CssSelector("#testId-search-wrapper > div > button")).Click();
      Thread.Sleep(1000);
      GreenHeartHelper.TakeEvidenceWithScreenshot("Búsqueda",
          "Se realiza la búsqueda de " + text + " correctamente", driver);
    }

    [TearDown]
    public async Task PublishToGreenHeartAsync()
    {
      bool testResult = true;
      string testName = TestContext.CurrentContext.Test.MethodName;
      long testElapsedTime = (long)(DateTime.Now - startTime).TotalMilliseconds;

      if (TestContext.CurrentContext.Result.Message != null)
      {
        string formatedError = System.Text.RegularExpressions.Regex.Replace(TestContext.CurrentContext.Result.Message, @"\t|\n|\r", "");
        GreenHeartHelper.TakeEvidence("Error", formatedError);
        testResult = false;
      }

      switch (testName)
      {
        case "A_validarDisponibilidadFalabella":
          {
            await GreenHeartHelper.PublishAsync(testResult, testName, "cgordillo", 3000, 12000,
                testElapsedTime, 2.5, DevEnvironment.DEV, GsqaService.TIA,
                "GreenHeart");
            break;
          }
        case "B_busquedaFalabella":
          {
            await GreenHeartHelper.PublishAsync(testResult, testName, "cgordillo", 2000, 8000,
                testElapsedTime, 2.0, DevEnvironment.DEV, GsqaService.TIA,
                "GreenHeart");
            break;
          }
      }
    }

    [OneTimeTearDown]
    public void KillDriver()
    {
      driver.Close();
    }
  }
}
